web/src/pages/Environments/EnvironmentContent.tsx

```diff
@@ -0,0 +1,31 @@
+import {useCallback, useState} from 'react';
+import SearchInput from '../../components/SearchInput';
+import {IEnvironment} from '../../redux/apis/TraceTest.api';
+import EnvironmentActions from './EnvironmentActions';
+import {EnvironmentState} from './EnvironmentState';
+import EnvironmentList from './EnvList';
+import * as S from './Envs.styled';
+import {EnvsModal} from './EnvsModal';
+
+const EnvironmentContent: React.FC = () => {
+  const [state, setState] = useState<EnvironmentState>({query: '', dialog: false, environment: undefined});
```

Comment:
This feels a little convoluted, why is the use for having everything in a single state vs separated?

Reply:
OK! I have separated the states ✅